cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/services/ApplicationLifecycleService.java

```diff
@@ -337,7 +337,10 @@ public ApplicationWithPrograms updateApp(ApplicationId appId, AppRequest appRequ
     // ownerAdmin.getImpersonationPrincipal will give the owner which will be impersonated for the application
     // irrespective of the version
     if (!Objects.equals(ownerAdmin.getImpersonationPrincipal(appId), appRequest.getOwnerPrincipal())) {
-      throw new ConflictException(String.format("Updating %s is not supported.", Constants.Security.PRINCIPAL));
+      throw new ConflictException(String.format("The specified %s: '%s' is not same as existing one. " +
```

Comment:
The message in AppLifeCycleService and FileStreamAdmin is different than ApplicationVerificationStage because here the message is displayed when the user is updating the owner principal whereas in ApplicationVerificationStage its displayed when the user deploy the same app with different owner hence that one says already exists and these don't. It looks weird to say entity already exists when someone is trying to update a property. Don't you think so ?